pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -311,19 +316,44 @@ func (s *Server) initInprocessAnalysisController(args *PilotArgs) error {
 	processing := components.NewProcessing(processingArgs)
 
 	s.addStartFunc(func(stop <-chan struct{}) error {
-		if err := processing.Start(); err != nil {
-			return err
-		}
+		go leaderelection.
+			NewLeaderElection(args.Namespace, args.PodName, leaderelection.StatusController, s.kubeClient).
+			AddRunFunction(func(stop <-chan struct{}) {
+				if err := processing.Start(); err != nil {
+					log.Fatalf("Error starting Background Analysis: %s", err)
+				}
 
-		go func() {
-			<-stop
-			processing.Stop()
-		}()
+				go func() {
+					<-stop
+					processing.Stop()
+				}()
+			}).Run(stop)
 		return nil
 	})
 	return nil
 }
 
+func (s *Server) initStatusController(args *PilotArgs) {
+	s.addTerminatingStartFunc(func(stop <-chan struct{}) error {
+		leaderelection.
+			NewLeaderElection(args.Namespace, args.PodName, leaderelection.StatusController, s.kubeClient).
+			AddRunFunction(func(stop <-chan struct{}) {
+				(&status.DistributionController{QPS: float32(features.StatusQPS), Burst: features.StatusBurst}).
+					Start(s.kubeConfig, args.Namespace, stop)
+			}).Run(stop)
+		return nil
+	})
+	s.statusReporter = &status.Reporter{
+		UpdateInterval: time.Millisecond * 500, // TODO: use args here?
+		PodName:        args.PodName,
+	}
+	s.addTerminatingStartFunc(func(stop <-chan struct{}) error {
+		s.statusReporter.Start(s.kubeConfig, args.Namespace, s.configController, stop)
```

Comment:
this doesn't seem to properly clean up despite the `addTerminatingStartFunc`

Reply:
Agreed.  As far as I can tell, the stop channel is not actually closed before the process exits.